dashboard/app/models/school_info.rb

```diff
@@ -290,4 +290,22 @@ def public_school?
   def charter_school?
     school_type.eql? SCHOOL_TYPE_CHARTER
   end
+
+  # Decides whether the school info is complete enough to stop bugging the
+  # teacher for additional information every week.  Different from complete
+  # record validation.
+  def complete?
+    return true unless school_id.nil?
+    return false if country.nil?
+    return true unless usa?
+    return true if [
+      SchoolInfo::SCHOOL_TYPE_HOMESCHOOL,
+      SchoolInfo::SCHOOL_TYPE_AFTER_SCHOOL,
+      SchoolInfo::SCHOOL_TYPE_ORGANIZATION,
+      SchoolInfo::SCHOOL_TYPE_OTHER,
+    ].include?(school_type)
+
+    # Given we got past above cases, school name is sufficient
+    !!school_name
```

Comment:
should this be `!school_name.blank?`?

Reply:
Maybe? We moved this as-is [from school_info_interstitial_helper.rb](https://github.com/code-dot-org/code-dot-org/pull/28849/files#diff-d0658a8986d877308a56d3714e9abd87L72).  We'll take a look at whether it makes sense to change this.